plugins/modules/accesspoint_workflow_manager.py

```diff
@@ -0,0 +1,818 @@
+                                .format(eachap["led_status"]))
+
+            if eachap.get("management_ip_address"):
+                if not self.is_valid_ipv4(eachap["management_ip_address"]):
+                    errormsg.append("management_ip_address: Invalid Management IP Address '{0}' in playbook"\
+                                    .format(eachap["management_ip_address"]))
+
+            if eachap.get("ap_name"):
+                param_spec = dict(type = "str", length_max = 32)
+                validate_str(eachap["ap_name"], param_spec, "ap_name",
+                                errormsg)
+
+            if eachap.get("ap_name_new"):
+                param_spec = dict(type = "str", length_max = 32)
+                validate_str(eachap["ap_name_new"], param_spec, "ap_name_new",
+                                errormsg)
+
+            if eachap.get("location"):
+                param_spec = dict(type = "str", length_max = 255)
+                validate_str(eachap["location"], param_spec, "location",
+                                errormsg)
+
+            if len(errormsg) > 0:
+                self.log("Invalid parameters in playbook file: '{0}' ".format(str("\n".join(errormsg))), "ERROR")
+                self.module.fail_json(msg=str("\n".join(errormsg)))
+            else:
+                self.validated_config = valid_param
+                self.msg = "Successfully validated playbook config params: {0}".format(str(valid_param))
+                self.log(self.msg, "INFO")
+                self.status = "success"
+                return self
+
+        except Exception as e:
+            self.log("Invalid Param provided in playbook Yml File. {0}".format(str(e)), "ERROR")
+            self.msg = "Invalid parameters in playbook: {0}".format(str("\n".join(errormsg)))
+            self.status = "failed"
+            return self
+
+    def get_want(self, ap_config):
+        """
+        Get all site-related information from the playbook needed for creation/updation/deletion of site in Cisco Catalyst Center.
+        Parameters:
+            self (object): An instance of a class used for interacting with Cisco Catalyst Center.
+            config (dict): A dictionary containing configuration information.
+        Returns:
+            self (object): An instance of a class used for interacting with Cisco Catalyst Center.
+        Description:
+            Retrieves all site-related information from playbook that is
+            required for creating a site in Cisco Catalyst Center. It includes
+            parameters such as 'site_params' and 'site_name.' The gathered
+            information is stored in the 'want' attribute for later reference.
+        """
+        for key,value in ap_config.items():
+            if key not in ("device_fields", "ap_selected_field"):
+                self.want[key] = value
+        self.log("Desired State (want): {0}".format(str(self.want)), "INFO")
```

Comment:
create a new dict here and store the input and then assign it to self.want - 

```
want = {}
for key,value in ap_config.items():
    if key not in ("device_fields", "ap_selected_field"):
        want[key] = value
self.want = want 
```

Reply:
updated